packages/components/src/drop-zone/animators/maybe-fade.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { useReducedMotion } from '@wordpress/compose';
+
+/**
+ * Handles fade-in/fade-out animation for its children if `reducedMotion` is disabled,
+ * if not, just shows/hides the children without any animation.
+ *
+ * Relies on the animations being defined in CSS styles.
+ */
+export const MaybeFade: React.FC< {
```

Comment:
I think this wrapper component is not necessary, see comments below.

Reply:
I feel like we should just settle for global classes with proper names, e.g. `.wp-animation_fade`, something like that. If we go that way, we should be clear on the right naming pattern from the start.

I wanted to experiment with this but you beat me to it (thank you for getting this started!).

Reply:
Sounds good! I'll work on simplifying this to use transition and the media query for reduced motion. Thanks, folks!

Reply:
> I feel like we should just settle for global classes with proper names, e.g. .wp-animation_fade

@DaniGuardiola I didn't try to generalize in this iteration, I think the fade transition is simple enough and very specific to this component, but let me know if you think otherwise.